README.md

```diff
@@ -0,0 +1,45 @@
+# Clickbait Filter
+
+This project is an optional challenge of the mandatory course 
+"Machine Learning for Natural Language Understanding" of my NLP master's degree
+program at Trier University in the winter semester 22/23. 
+
+## Task
+
+The Task was to train a clickbait filter to classify clickbait articles 
+by their headline. I could freely decide how to prepare the data and which ML 
+model to use for classification.
+
+The challenge was considered passed if the model performs better than professor's 
+baseline (a simple classifier; F1 ~0.89). 
+
+## Dataset
+The data consists of two files, a text file with clickbait 
+headlines and one with headlines from news sources. 
+The hold out dataset is organized the same way.
+
+I'm not allowed to publish the train and validation datasets since they are a property of
+Computerlinguistik und Digital Humanities Department of the University of Trier.
+
+## Results
+
+I implemented an [LSTM model](https://pytorch.org/docs/stable/generated/torch.nn.LSTM.html) (Raschka, 2022, p.499)
+with [dropouts](https://pytorch.org/docs/stable/generated/torch.nn.Dropout.html) using PyTorch library (`./utils/models.py`)
+It showed a quite good result on the validation set: ___F1-Score = 96.2%___ (`./notebooks/validation_and_examples.ipynb`) which is however can be easily overcome with Transformer architecture.
+
+## How to use
+`git clone https://github.com/bourgeois-radical/clickbait-detection.git`
+
+ _ClickbaitClassifier_ class (`./utils/showing_results.py`) 
+provides a dunder-method which classifies every English sentence you give. 
```

Comment:
I would add code examples from `validation_and_examples.ipynb` notebook. Something like


`check_clickbait = ClickbaitClassifier(model_testing, preprocesser, vocab)

some_sentences = ["How I Made $1 Million Last Week From Clicking Here", "Russia-Ukraine war: Germany blocks offer of tanks for Ukraine",
                 "What Happened When I Clicked Here And Asked Her On A Date", "Kylian Mbappe's affectionate gesture to Cristiano that he's never made with Messi" ]

for example_sentence in some_sentences:
    print(f"'{example_sentence}' is clickbait with probability = {check_clickbait(example_sentence)*100:.2f}%\n")
    
>>> 'How I Made $1 Million Last Week From Clicking Here' is clickbait with probability = 99.96%
>>>'Russia-Ukraine war: Germany blocks offer of tanks for Ukraine' is clickbait with probability = 0.00%`